gap/Parser.gi

```diff
@@ -438,13 +438,19 @@ InstallGlobalFunction( AutoDoc_Parser_ReadFiles,
                                  or PositionSublist( temp_curr_line, "@EndLogSession" ) <> fail then
                 break;
             fi;
-            #! @DONT_SCAN_NEXT_LINE
-            temp_pos_comment := PositionSublist( temp_curr_line, "#!" );
-            if temp_pos_comment <> fail then
-                temp_curr_line := temp_curr_line{[ temp_pos_comment + 2 .. Length( temp_curr_line ) ]};
-                if Length( temp_curr_line ) >= 1 and temp_curr_line[ 1 ] = ' ' then
-                    Remove( temp_curr_line, 1 );
+            incorporate_this_line := pt_mode;
+            if not pt_mode then
+                #! @DONT_SCAN_NEXT_LINE
+                temp_pos_comment := PositionSublist( temp_curr_line, "#!" );
+                if temp_pos_comment <> fail then
+                    incorporate_this_line := true;
+                    temp_curr_line := temp_curr_line{[ temp_pos_comment + 2 .. Length( temp_curr_line ) ]};
+                    if Length( temp_curr_line ) >= 1 and temp_curr_line[ 1 ] = ' ' then
+                        Remove( temp_curr_line, 1 );
+                    fi;
                 fi;
+            fi;
+            if incorporate_this_line then
```

Comment:
I am not very familiar with this code, so @sebasguts should review this part of the PR.

Reply:
@sebasguts could you review this PR? I have now updated it (mostly the tests)